python/ray/serve/_private/router.py

```diff
@@ -633,14 +646,22 @@ async def assign_request(
                     )
                     replica_result.add_done_callback(callback)
 
-                return replica_result
+                if request_meta.is_streaming:
+                    # For streaming requests, return the async iterator directly
+                    logger.info("Returning streaming replica result.")
+                    return replica_result.to_object_ref_gen()
```

Comment:
@alexeykudinkin in the streaming case, we don't have an opportunity to await replica_result here. Hence, even with this change, our initial problem of the cancel not propagating is not solved directly.

Reply:
Wait, why can't we do `to_object_ref_async` here?

Reply:
That function is only for non non-streaming case https://github.com/anyscale/rayturbo/blob/4c0fe1e433eea6db3fdb2dc21c5858dc05d067f0/python/ray/serve/_private/replica_result.py#L178-L191

The app will only have access to the first response and cannot use it to iterate over the next elements.